lib/average_end_of_day_loan_balance/main.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import asyncio
+import os
+
+from datetime import datetime, timedelta
+from typing import Optional
+
+import pandas as pd  # type: ignore
+
+from pngme.api import AsyncClient
+
+
+async def get_average_end_of_day_balance(
+    api_client: AsyncClient,
+    user_uuid: str,
+    utc_starttime: datetime,
+    utc_endtime: datetime,
+) -> Optional[float]:
+    """Calculates the average end-of-day total balance for a user across all
+       of their accounts for a given time window.
+
+       If no balance data was found, return None.
+
+    Args:
+        api_client: Pngme Async API client
+        user_uuid: the Pngme user_uuid for the mobile phone user
+        utc_starttime: the UTC time to start the time window
+        utc_endtime: the UTC time to end the time window
+
+    Returns:
+        the average end-of-day total balance over each window
+    """
+
+    institutions = await api_client.institutions.get(user_uuid=user_uuid)
+
+    # subset to only fetch data for institutions known to contain loan-type accounts for the user
+    institutions_w_loan = [
+        inst for inst in institutions if "loan" in inst.account_types
+    ]
+
+    # Construct timerange since beginning of time
+    # as default /balances endpoint only returns the latest balance
+    today = datetime.today()
+    all_pages_utc_endtime = datetime(today.year, today.month, today.day)
+    all_pages_starttime = all_pages_utc_endtime - timedelta(days=1e5)
+
+    inst_coroutines = [
+        api_client.balances.get(
+            user_uuid=user_uuid,
+            institution_id=institution.institution_id,
+            utc_starttime=all_pages_starttime,
+            utc_endtime=all_pages_utc_endtime,
+            account_types=["loan"],
+        )
+        for institution in institutions_w_loan
+    ]
+    r = await asyncio.gather(*inst_coroutines)
+
+    record_list = []
+    for ix, inst_list in enumerate(r):
+        institution_id = institutions[ix].institution_id
+        record_list.extend(
+            [
+                dict(transaction, institution_id=institution_id)
+                for transaction in inst_list
+            ]
+        )
+    # consider the sum of balances to be zero, if no data is present
+    if len(record_list) == 0:
+        return None
+
+    balances_df = pd.DataFrame(record_list)
+
+    # 2. Sort and create a column for day, filter by time window
```

Comment:
What is this enumerating?  I don't see a '#1'

Reply:
I literally copied the code for balances and did not notice the comments. I do not like enumerations in comments because of this kind of silly issues. Fixed.